app/commands/markdown/render_html.rb

```diff
@@ -21,9 +21,19 @@ def link(node)
       out('<a href="', node.url.nil? ? '' : escape_href(node.url), '" target="_blank"')
       out(' title="', escape_html(node.title), '"') if node.title.present?
       out(' rel="nofollow"') if nofollow_links
+      out(link_tooltip_attributes(node))
       out('>', :children, '</a>')
     end
 
+    def link_tooltip_attributes(node)
+      link_match = %r{^(?<url>https://exercism.io)?/tracks/(?<track>[a-zA-Z0-9_-]+)/(?<type>concept|exercise)s/(?<slug>[a-zA-Z0-9_-]+)}.match(node.url) # rubocop:disable Layout/LineLength
+      return unless link_match
+
+      endpoint = Exercism::Routes.send("tooltip_track_#{link_match[:type]}_path", track_id: link_match[:track],
```

Comment:
1. If we're going to arbitarily send methods like this, we need to ensure that `link_match[:type]` is an expected value, else we open ourselves up to injection. So I'd guard that it's either "exercise" or "concept" above this.
2. I don't **think** you need the `track_id` and `id` keys here. I think you can just pass the values

Reply:
Will do